packages/@react-spectrum/dropzone/docs/DropZone.mdx

```diff
@@ -0,0 +1,338 @@
+import ChevronRight from '@spectrum-icons/workflow/ChevronRight';
+
+```jsx import
+import {DropZone} from '@react-spectrum/dropzone';
+import {Heading} from '@react-spectrum/text';
+import {Content} from '@react-spectrum/view';
+import {IllustratedMessage} from '@react-spectrum/illustratedmessage';
+import {Text} from 'react-aria-components';
+import {FileTrigger} from 'react-aria-components';
+import {Button} from '@react-spectrum/button';
+```
+
+---
+category: Drag and drop
+---
+
+# DropZone
+
+<PageDescription>{docs.exports.DropZone.description}</PageDescription>
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['DropZone']}
+  since="3.0.0" />
+
+## Example
+
+```tsx example
+import Upload from '@spectrum-icons/illustrations/Upload';
+
+<DropZone
+  maxWidth="size-3000">
+  <IllustratedMessage>
+    <Upload />
+    <Heading>
+      <Text slot="label">
+        Drag and drop your file
+      </Text>
+    </Heading>
+  </IllustratedMessage>
+</DropZone>
+```
+
+
+## Content
+
+A drop zone accepts an [IllustratedMessage](IllustratedMessage.html) as a child which is comprised of three areas:  an illustration, a title, and a body. Each of these sections can be populated by providing the following components to the IllustratedMessage as children: svg, [Heading](Heading.html) (title), and [Content](Content.html) (body). A [FileTrigger](../react-aria/FileTrigger.html) is commonly paired with a DropZone to allow a user to choose files from their device. 
+
+```tsx example
+<DropZone
+  maxWidth="size-3000">
+  <IllustratedMessage>
+    <Upload />
+    <Heading>
+      <Text slot="label">
+        Drag and drop here
+      </Text>
+    </Heading>
+    <Content>
+      <FileTrigger>
+        <Button variant="primary">Browse</Button>
+      </FileTrigger>
+    </Content>
+  </IllustratedMessage>
+</DropZone>
+```
+
+### Accessibility
+
+A visual label should be provided to `DropZone` using a `Text` element with a `label` slot. If it is not provided, then an `aria-label` or `aria-labelledby` prop must be passed to identify the visually hidden button to assistive technology. 
+
+## Events
+
+`DropZone` supports user interactions via mouse, keyboard, and touch. You can handle all of these via the `onDrop` prop. In addition, the `onDropEnter`, `onDropMove`, and `onDropExit` events are fired as the user interacts with the dropzone. 
+
+The following example uses an `onDrop` handler to update the filled status stored in React state.
+
+```tsx example
+function Example() {
+  let [isFilled, setIsFilled] = React.useState(false);
+
+  return (
+    <>
+      <Draggable />
+      <DropZone
+        maxWidth="size-3000"
+        isFilled={isFilled}
+        onDrop={() => setIsFilled(true)}>
+        <IllustratedMessage>
+          <Upload />
+          <Heading>
+            <Text slot="label">
+              Drag and drop here
+            </Text>
+          </Heading>
+        </IllustratedMessage>
+      </DropZone>
+    </>
+  );
+}
+```
```

Comment:
IMO this example might be hard for the user to understand since it requires a full drop operation + a hover/keyboard drop focus to see the results of the `onDrop`. Perhaps you could mimic one of your story examples that print out the filename/text of the dropped item upon drop instead?